servo/connectors/kubernetes.py

```diff
@@ -1439,6 +1439,16 @@ async def delete(self, options:kubernetes_asyncio.client.V1DeleteOptions = None)
                 body=options,
             )
 
+    async def scale_to_zero(self) -> None:
+        """this is used instead of 'delete' as the 'destroy' operation to handle 'on_failure: destroy'.
+        Since the Deployment object is used as a wrapper around an existing k8s object that we did not create,
+        it shouldn't be destroyed. Instead, the deployments pods are destroyed by scaling it to 0 replicas.
+        """
```

Comment:
Commenting about the specifics of the `on_failure` semantics here feels like an irrelevant detail about the motivation for introducing the operation is leaking in and violating layering. The `Deployment` model doesn't know or care about `on_failure` -- that is a downstream concept.

Scaling the replicas to zero seems like a perfectly reasonable operation to have as a simple shortcut in the general `Deployment` API though.

One thought to consider on naming off the top of my head is call the operation `shutdown`. I'm not sure if the succinctness is better than the explicit clarity of `scale_to_zero` but throwing it out there as a thought